src/functions/awesome_memgraph_functions.hpp

```diff
@@ -1026,12 +966,9 @@ template <typename TypedValueT, typename FunctionContextT>
 TypedValueT Id(const TypedValueT *args, int64_t nargs, const FunctionContextT &ctx) {
   FType<TypedValueT, Or<Null, Vertex, Edge>>("id", args, nargs);
   const auto &arg = args[0];
-  if (arg.IsNull()) {
+  if (arg.IsNull() || arg.IsVertex()) {
     return TypedValueT(ctx.memory);
   }
-  if (arg.IsVertex()) {
-    return TypedValueT(static_cast<int64_t>(arg.ValueVertex().CypherId()), ctx.memory);
-  }
   return TypedValueT(static_cast<int64_t>(arg.ValueEdge().CypherId()), ctx.memory);
 }
```

Comment:
Why not to modify the `FType` on line 967 to simply "exclude" the vertices from this function?